.ci/ci-build.sh

```diff
@@ -29,16 +29,54 @@ tarball=`ls -1 cc-oci-runtime-*.tar.xz`
 chronic tar xvf "$tarball"
 tarball_dir=${tarball%.tar.xz}
 
+./autogen.sh
+
 # We run configure and make check in an exploded make dist tarball to make sure
 # we distribute the necessary files for both building and testing.
 # We also do an out of tree build to check srcdir vs builddir discrepancy.
-mkdir travis_build
-(cd travis_build && \
- ../"$tarball_dir"/configure \
-     --sysconfdir=/etc --localstatedir=/var --prefix=/usr \
-     --enable-cppcheck \
-     --enable-valgrind --disable-valgrind-helgrind --disable-valgrind-drd \
-     --disable-silent-rules \
-     --disable-docker-tests \
+mkdir ci_build
+
+configure_opts=""
+configure_opts+=" --sysconfdir=/etc"
+configure_opts+=" --localstatedir=/var"
+configure_opts+=" --prefix=/usr"
+configure_opts+=" --enable-cppcheck"
+configure_opts+=" --enable-valgrind"
+configure_opts+=" --disable-valgrind-helgrind"
+configure_opts+=" --disable-valgrind-drd"
+configure_opts+=" --disable-silent-rules"
+
+nested=$(cat /sys/module/kvm_intel/parameters/nested 2>/dev/null \
+    || echo N)
+
+echo "INFO: Nested kvm available: $nested"
+
+if [ "$nested" = "Y" ]
+then
+    configure_opts+=" --enable-docker-tests"
+    configure_opts+=" --enable-functional-tests"
+else
+    configure_opts+=" --disable-docker-tests"
+    configure_opts+=" --disable-functional-tests"
+fi
+
+(cd ci_build && \
+ ../"$tarball_dir"/configure "$configure_opts" \
  && make -j5 CFLAGS=-Werror \
  && make check)
+
+# go checks
+export go_packages=$(go list ./... |\
+    grep -v cc-oci-runtime/vendor |\
+    sed -e 's#.*/cc-oci-runtime/#./#')
+
+go list -f '{{.Dir}}/*.go' "$go_packages" |\
+    xargs -I % bash -c "misspell -error %"
+
+go vet "$go_packages"
```

Comment:
Quoting variables isn't always the right thing to do :)

```
$ go_packages=$(go list ./... | grep -v cc-oci-runtime/vendor | sed -e 's#.*/cc-oci-runtime/#./#')
$ echo $go_packages 
./proxy ./proxy/api ./tests/mock
$ go vet $go_packages
$ go vet "$go_packages"
can't load package: package ./proxy
./proxy/api
./tests/mock: open /home/damien/clear/sources/cc-oci-runtime/proxy
./proxy/api
./tests/mock: no such file or directory
```

(go vet will receive the whole list as a single argument instead of 3)